src/features/user_profile/components/SettingsForm/ReferenceAreaInfo/RefAreaSearchBar.tsx

```diff
@@ -0,0 +1,132 @@
+import { useAction, useAtom } from '@reatom/npm-react';
+import { SelectItem } from '@konturio/ui-kit';
+import { useEffect } from 'react';
+import { SearchInput } from '~components/Search/SearchInput/SearchInput';
+import { i18n } from '~core/localization';
+import { useSearchMenu } from '~utils/hooks/useSearchMenu';
+import {
+  resetSearchRefAreaAtom,
+  searchRefAreaAtom,
+  searchAction,
+} from '~features/user_profile/components/SettingsForm/ReferenceAreaInfo/searchRefAreaAtoms';
+import { updateReferenceArea } from '~core/api/features';
+import { setReferenceArea } from '~core/shared_state/referenceArea';
+import { store } from '~core/store/store';
+import { notificationServiceInstance } from '~core/notificationServiceInstance';
+import s from './RefAreaSearchBar.module.css';
+
+export const RefAreaSearchBar = () => {
```

Comment:
Ref can be confused with react `ref`, but anyway abbreviations are not desirable